flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,183 @@
+
+        case GT:
+          return convertComparisonExpression(Expressions::greaterThan, call);
+
+        case GT_EQ:
+          return convertComparisonExpression(Expressions::greaterThanOrEqual, call);
+
+        case EQ:
+          return convertComparisonExpression(Expressions::equal, call);
+
+        case NOT_EQ:
+          return convertComparisonExpression(Expressions::notEqual, call);
+
+        case IN:
+          List<ResolvedExpression> args = call.getResolvedChildren();
+          FieldReferenceExpression field = (FieldReferenceExpression) args.get(0);
+          List<ResolvedExpression> values = args.subList(1, args.size());
+
+          List<Object> inputValues = values.stream().filter(
+              expression -> {
+                if (expression instanceof ValueLiteralExpression) {
+                  return !((ValueLiteralExpression) flinkExpression).isNull();
+                }
+
+                return false;
+              }
+          ).map(expression -> {
+            ValueLiteralExpression valueLiteralExpression = (ValueLiteralExpression) expression;
+            return valueLiteralExpression.getValueAs(valueLiteralExpression.getOutputDataType().getConversionClass())
+                .get();
+          }).collect(Collectors.toList());
+          return Optional.of(Expressions.in(field.getName(), inputValues));
+
+        case NOT:
+          Optional<Expression> child = convert(call.getResolvedChildren().get(0));
+          if (child.isPresent()) {
+            return Optional.of(Expressions.not(child.get()));
+          }
+
+          return Optional.empty();
+
+        case AND: {
+          return convertLogicExpression(Expressions::and, call);
+        }
+
+        case OR: {
+          return convertLogicExpression(Expressions::or, call);
+        }
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                             CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    Optional<Expression> left = convert(args.get(0));
+    Optional<Expression> right = convert(args.get(1));
+    if (left.isPresent() && right.isPresent()) {
+      return Optional.of(function.apply(left.get(), right.get()));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertComparisonExpression(BiFunction<String, Object, Expression> function,
+                                                                  CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    FieldReferenceExpression fieldReferenceExpression;
+    ValueLiteralExpression valueLiteralExpression;
+    if (literalOnRight(args)) {
+      fieldReferenceExpression = (FieldReferenceExpression) args.get(0);
+      valueLiteralExpression = (ValueLiteralExpression) args.get(1);
+    } else {
```

Comment:
Q:  is it possible that we have two literals in both left and right side ? For example,  someone may write the SQL: 
`SELECT * FROM test where 1 > 0`,  then we could not think that the left MUST be a field and the right MUST be a `value` in this `else` block.   

We have done the similar work in our own branch before,  you could see the PR: https://github.com/generic-datalake/iceberg-poc/pull/2/files#diff-86160616589acf1dd526b10b73418a46fe60f9e5e5ab6946a4ea3c8f019542f5R123

Reply:
I think it's better to add an unit tests to address this case.

Reply:
I agree. It is best to validate both child types in each case and only convert if they match what is expected.

Reply:
I add the test case [literal on right](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java#L133),[ literal on left](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java#L152) ,[2 literals](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java#L416)